pkg/services/ngalert/provisioning/mute_timings.go

```diff
@@ -179,26 +179,31 @@ func (svc *MuteTimingService) UpdateMuteTiming(ctx context.Context, mt definitio
 
 // DeleteMuteTiming deletes the mute timing with the given name in the given org. If the mute timing does not exist, no error is returned.
 func (svc *MuteTimingService) DeleteMuteTiming(ctx context.Context, name string, orgID int64, provenance definitions.Provenance, version string) error {
-	target := definitions.MuteTimeInterval{MuteTimeInterval: config.MuteTimeInterval{Name: name}, Provenance: provenance}
-	// check that provenance is not changed in an invalid way
-	storedProvenance, err := svc.provenanceStore.GetProvenance(ctx, &target, orgID)
+	revision, err := svc.configStore.Get(ctx, orgID)
 	if err != nil {
 		return err
 	}
-	if err := svc.validator(storedProvenance, models.Provenance(provenance)); err != nil {
-		return err
+
+	existing, idx := getMuteTiming(revision, name)
+	if idx == -1 {
+		svc.log.FromContext(ctx).Debug("Time interval was not found. Skip deleting", "name", name)
+		return nil
 	}
 
-	revision, err := svc.configStore.Get(ctx, orgID)
+	target := definitions.MuteTimeInterval{MuteTimeInterval: existing, Provenance: provenance}
+	// check that provenance is not changed in an invalid way
+	storedProvenance, err := svc.provenanceStore.GetProvenance(ctx, &target, orgID)
 	if err != nil {
 		return err
 	}
-
-	if revision.cfg.AlertmanagerConfig.MuteTimeIntervals == nil {
-		return nil
+	if err := svc.validator(storedProvenance, models.Provenance(provenance)); err != nil {
+		return err
 	}
-	if isMuteTimeInUse(name, []*definitions.Route{revision.cfg.AlertmanagerConfig.Route}) {
-		return ErrTimeIntervalInUse.Errorf("")
+
+	if isMuteTimeInUseInRoutes(name, revision.cfg.AlertmanagerConfig.Route) {
+		ns, _ := svc.ruleNotificationsStore.ListNotificationSettings(ctx, models.ListNotificationSettingsQuery{OrgID: orgID, TimeIntervalName: existing.Name})
+		// ignore error here because it's not important
+		return MakeErrTimeIntervalInUse(true, maps.Keys(ns))
 	}
 	for i, existing := range revision.cfg.AlertmanagerConfig.MuteTimeIntervals {
```

Comment:
NIT: This loop is no longer necessary since you have the `idx` from `getMuteTiming`.